src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenChatHandler.java

```diff
@@ -45,30 +41,30 @@ public void notifyDeath(final DamageSource damageSource)
         if (citizen.getCitizenColonyHandler().getColony() != null && citizen.getCitizenData() != null)
         {
             final IJob<?> job = citizen.getCitizenJobHandler().getColonyJob();
-            MutableComponent contentComponent;
+            Component nameComponent;
             if (job != null)
             {
-                contentComponent = Component.translatable(
-                  TranslationConstants.WORKER_DIED,
+                nameComponent = Component.translatable(
+                  TranslationConstants.WORKER_FULL,
                   Component.translatable(job.getJobRegistryEntry().getTranslationKey()),
-                  citizen.getCitizenData().getName(),
-                  Math.round(citizen.getX()),
-                  Math.round(citizen.getY()),
-                  Math.round(citizen.getZ()),
-                  Component.translatable(damageSource.msgId));
+                  citizen.getCitizenData().getName());
             }
             else
             {
-                contentComponent = Component.translatable(
-                  TranslationConstants.COLONIST_DIED,
-                  citizen.getCitizenData().getName(),
-                  Math.round(citizen.getX()),
-                  Math.round(citizen.getY()),
-                  Math.round(citizen.getZ()),
-                  Component.translatable(damageSource.msgId));
+                nameComponent = Component.translatable(
+                  TranslationConstants.COLONIST_FULL,
+                  citizen.getCitizenData().getName());
             }
-
-            MessageUtils.format(contentComponent)
+            String result = citizen.getCombatTracker().getDeathMessage().getString().replace(citizen.getDisplayName().getString(), nameComponent.getString());
+            if (isAlphanumeric(result.charAt(result.length() - 1))) {
+                result += "!";
+            }
+            result += " ";
+            result += Component.translatable(TranslationConstants.COLONIST_GRAVE_LOCATION,
```

Comment:
You should use literal components to add extra parts to your string, (like the "!" and " ").
Right now you are calling `.getString()` at the end, which mean it will opt out of any translation functionality since you're converting it to a string on the server side, meaning it's always English.